### PR TITLE
[repo] How to maintain PublicApi files

### PR DESCRIPTION
**Pull request by ysolomchenko**
Fixes https://github.com/open-telemetry/opentelemetry-dotnet/issues/2739

## Changes

Create explanation how to generate .publicApi

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -70,22 +70,28 @@ helper methods.
...
+    <AdditionalFiles Include=".publicApi\$(TargetFramework)\PublicAPI.Shipped.txt" />
+    <AdditionalFiles Include=".publicApi\$(TargetFramework)\PublicAPI.Unshipped.txt" />
+    </ItemGroup>
+    ```
 
-* Use
+4. Use
    [IntelliSense](https://docs.microsoft.com/visualstudio/ide/using-intellisense)
-   to update the publicApi files.
+   to utomatically update the public API files. When making changes,
```
Comment (rajkumar-rangaraj) [inline CONTRIBUTING.md:88]:
```suggestion
   to automatically update the public API files. When making changes,
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -70,22 +70,28 @@ helper methods.
...
+   [IntelliSense will suggest modifications](https://github.com/dotnet/roslyn-analyzers/issues/3322#issuecomment-591031429)
+   to the PublicAPI.Unshipped.txt file. After reviewing these changes,
```
Comment (rajkumar-rangaraj) [inline CONTRIBUTING.md:90]:
```suggestion
   to the `PublicAPI.Unshipped.txt` file. After reviewing these changes,
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -70,22 +70,28 @@ helper methods.
...
+   ensure they are reflected across all targeted frameworks by either:
+   * Manually copying the changes.
+   * Using Visual Studio's target framework dropdown (in the upper right corner)
```
Comment (rajkumar-rangaraj) [inline CONTRIBUTING.md:93]: Can you link to an article that explains how to get this done in Visual Studio?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -70,22 +70,28 @@ helper methods.
...
-<ItemGroup>
-  <AdditionalFiles Include=".publicApi\$(TargetFramework)\PublicAPI.Shipped.txt" />
-  <AdditionalFiles Include=".publicApi\$(TargetFramework)\PublicAPI.Unshipped.txt" />
-</ItemGroup>
-```
+    ```xml
+    <ItemGroup>
+    <AdditionalFiles Include=".publicApi\$(TargetFramework)\PublicAPI.Shipped.txt" />
+    <AdditionalFiles Include=".publicApi\$(TargetFramework)\PublicAPI.Unshipped.txt" />
+    </ItemGroup>
```
Comment (reyang) [inline CONTRIBUTING.md:83]:
```suggestion
    <ItemGroup>
      <AdditionalFiles Include=".publicApi\$(TargetFramework)\PublicAPI.Shipped.txt" />
      <AdditionalFiles Include=".publicApi\$(TargetFramework)\PublicAPI.Unshipped.txt" />
    </ItemGroup>
```

**Review by reyang — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -70,22 +70,30 @@ helper methods.
...
+4. Use
    [IntelliSense](https://docs.microsoft.com/visualstudio/ide/using-intellisense)
-   to update the publicApi files.
+   to automatically update the public API files. When making changes,
+   [IntelliSense will suggest modifications](https://github.com/dotnet/roslyn-analyzers/issues/3322#issuecomment-591031429)
+   to the `PublicAPI.Unshipped.txt` file. After reviewing these changes,
+   ensure they are reflected across all targeted frameworks by either:
+   * Use the "Fix all occurrences in Project" Feature: Visual Studio's
+     IntelliSense offers a "Fix all occurrences in Project" option, making it easy
+     to apply changes across your project.
```
Comment (TimothyMothra) [inline CONTRIBUTING.md:94]: nit: Can this be written another way to avoid specifying "Fix all occurrences in Project" twice?

**Review by TimothyMothra — APPROVED** [review]
LGTM. I left one more minor comment

**Review by rajkumar-rangaraj — APPROVED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -70,22 +70,30 @@ helper methods.
...
 #### How to enable and configure
 
-* Create a folder in your project called `.publicApi` with the frameworks that
+1. Create a folder in your project called `.publicApi` with the frameworks that
   as folders you target.
-* Create two files called `PublicAPI.Shipped.txt` and `PublicAPI.Unshipped.txt`
+2. Create two files called `PublicAPI.Shipped.txt` and `PublicAPI.Unshipped.txt`
    in each framework that you target.
-* Add the following lines to your csproj:
+3. Add the following lines to your csproj:
```
Comment (CodeBlanch) [inline CONTRIBUTING.md:77]: Let's remove this step.

Everything should be handled automatically across the repo:

* https://github.com/open-telemetry/opentelemetry-dotnet/blob/37535a5607ee7e4056c0e274ec01d1e0111a64be/build/Common.prod.props#L49-L50

* https://github.com/open-telemetry/opentelemetry-dotnet/blob/37535a5607ee7e4056c0e274ec01d1e0111a64be/build/Common.prod.props#L102-L109

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -70,22 +70,30 @@ helper methods.
 
 #### How to enable and configure
 
-* Create a folder in your project called `.publicApi` with the frameworks that
+1. Create a folder in your project called `.publicApi` with the frameworks that
```
Comment (CodeBlanch) [inline CONTRIBUTING.md:73]: What needs to be done here is a bit more nuanced 🤣 

Here are the rules:

* If you are NOT using experimental APIS...

  * If your API is the same for all targets...

    You only need two files: `.publicApi/PublicAPI.Shipped.txt` and `.publicApi/PublicAPI.Unshipped.txt`

  * If your APIs differ between targets...

    a) Put all the shared APIs into these two files: `.publicApi/PublicAPI.Shipped.txt` and `.publicApi/PublicAPI.Unshipped.txt`
  
    b) Create framework-specific files for the API differences: `.publicApi/net462/PublicAPI.Shipped.txt` and `.publicApi/net462/PublicAPI.Unshipped.txt`

    Example: https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Api/.publicApi/Stable

* If you are using experimental APIs...

  * Follow the rules as above except create an additional layer in your folder structure: `.publicApi/Experimental/*` and `.publicApi/Stable/*`
  
    The `Experimental` folder should contain APIs which are `public` only in pre-release builds.

    Example: `https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry/.publicApi`

**Review by TimothyMothra — COMMENTED** [review]
Removing my approval. Please respond to Blanch's comments.

**Review by TimothyMothra — APPROVED** [review]
LGTM. Ship it! :)

**Pull request merged**